modules/islandora_simple_map_kml/islandora_simple_map_kml.module

```diff
@@ -0,0 +1,122 @@
+  $items = array();
+  $items['admin/islandora/tools/islandora_simple_map/kml'] = array(
+    'title' => 'KML',
+    'description' => 'Configure the KML settings.',
+    'page callback' => 'drupal_get_form',
+    'access arguments' => array('administer site configuration'),
+    'page arguments' => array('islandora_simple_map_kml_admin_settings_form'),
+    'type' => MENU_LOCAL_TASK,
+    'file' => 'includes/admin_form.inc',
+    'weight' => 10,
+  );
+
+  $items['islandora/object/%islandora_object/datastream/%islandora_datastream/islandora_simple_map_kml_mods'] = array(
+    'title' => 'Scrape MODS KML',
+    'page callback' => '_islandora_simple_map_kml_mods',
+    'page arguments' => array(2, 4),
+    'access callback' => '_islandora_simple_map_kml_mods_access_callback',
+    'access arguments' => array(2),
+    'type' => MENU_CALLBACK,
+    'file' => 'includes/utilities.inc',
+  );
+
+  $items['islandora/object/%islandora_object/datastream/%islandora_datastream/islandora_simple_map_kml_ddi'] = array(
+    'title' => 'Scrape DDI KML',
+    'page callback' => '_islandora_simple_map_kml_ddi',
+    'page arguments' => array(2, 4),
+    'access callback' => '_islandora_simple_map_kml_ddi_access_callback',
+    'access arguments' => array(2),
+    'type' => MENU_CALLBACK,
+    'file' => 'includes/utilities.inc',
+  );
+
+  return $items;
+}
+
+/**
+ * Access callback for the KML MODS callback.
+ */
+function _islandora_simple_map_kml_mods_access_callback($datastream) {
+  return $datastream->id == 'MODS' && islandora_datastream_access(ISLANDORA_VIEW_OBJECTS, $datastream);
+}
+
+/**
+ * Access callback for the KML scrape callback.
+ */
+function _islandora_simple_map_kml_ddi_access_callback($datastream) {
+  return $datastream->id == 'DDI' && islandora_datastream_access(ISLANDORA_VIEW_OBJECTS, $datastream);
+}
+
+/**
+ * Helper to invoke hooks to gather KML info for a given object.
+ *
+ * @param AbstractObject $object
+ *   The object for which to gather KML.
+ *
+ * @return array
+ *   An array of KML info.
+ */
+function islandora_simple_map_kml_get_kml(AbstractObject $object) {
+  $kml = array();
+
+  // Since KML is Google Maps-specific, check the map type being used.
+  if (variable_get('islandora_simple_map_map_type', ISLANDORA_SIMPLE_MAP_GOOGLE_MAPS) === ISLANDORA_SIMPLE_MAP_GOOGLE_MAPS) {
+    $kml = module_invoke_all('islandora_simple_map_kml_get_kml', $object);
+    drupal_alter('islandora_simple_map_kml_get_kml', $kml, $object);
+  }
+  return $kml;
+}
+
+/**
+ * Implements hook_islandora_simple_map_kml_get_kml().
+ */
+function islandora_simple_map_kml_islandora_simple_map_kml_get_kml(AbstractObject $object) {
+  $_split_parse = function ($to_parse) {
+    return drupal_map_assoc(array_filter(array_map('trim', explode("\r\n", $to_parse))));
+  };
+  $models = $_split_parse(variable_get('islandora_simple_map_kml_models', ''));
+  if ($models && !array_intersect($object->models, $models)) {
+    return;
```

Comment:
API doc says `hook_islandora_simple_map_kml_get_kml()` returns an array, so this should probably be `return array();`